lib/util/exec/docker/index.ts

```diff
@@ -256,3 +259,112 @@ export async function generateDockerCommand(
 
   return result.join(' ');
 }
+
+export async function volumePrune(
+  meta?: Record<string, string>
+): Promise<void> {
+  const cmdParts = ['docker', 'volume', 'prune', '--force'];
+
+  const kvPairs = Object.entries(meta);
+  const filterOptions = kvPairs.map(([k, v]) => `--filter label=${k}=${v}`);
+  cmdParts.push(...filterOptions);
+
+  const cmd = cmdParts.join(' ');
+  await rawExec(cmd, { encoding: 'utf-8' });
+}
+
+export async function volumeCreate(
+  name: string,
+  meta?: Record<string, string>
+): Promise<void> {
+  const cmdParts = ['docker', 'volume', 'create'];
+
+  if (meta) {
+    const kvPairs = Object.entries(meta);
+    const labelOptions = kvPairs.map(([k, v]) => `--label ${k}=${v}`);
+    cmdParts.push(...labelOptions);
+  }
+
+  cmdParts.push(name);
+
+  const cmd = cmdParts.join(' ');
+  await rawExec(cmd, { encoding: 'utf-8' });
+}
+
+let tmpId: string = null;
+
+export function getTmpCacheId(): string {
+  if (!tmpId) {
+    tmpId = cryptoRandomString({ length: 16 });
+  }
+  return tmpId;
+}
+
+export function resetTmpCacheId(): void {
+  tmpId = null;
+}
+
+export function getTmpCacheNs(): string {
+  const { dockerChildPrefix } = getAdminConfig();
+  return getContainerName('tmpdir_cache', dockerChildPrefix);
+}
+
+async function fixPermissionsBeforeDelete(entry: string): Promise<void> {
+  try {
+    if (await exists(entry)) {
+      const stats = await stat(entry);
+      if (stats.isDirectory()) {
+        await chmod(entry, '755');
+        const children = await readdir(entry);
+        for (const child of children) {
+          await fixPermissionsBeforeDelete(join(entry, child));
+        }
+      } else if (stats.isFile()) {
+        await chmod(entry, '644');
+      }
+    }
+  } catch (err) {
+    logger.debug({ err }, 'Permissions fixing error');
+  }
+}
+
+export async function removeDockerTmpCaches(): Promise<void> {
+  resetTmpCacheId();
+
+  const { binarySource, cacheDir, dockerCache } = getAdminConfig();
+  if (binarySource === 'docker') {
+    const cacheNamespace = getTmpCacheNs();
+    if (dockerCache === 'volume') {
+      logger.trace(`Deleting Docker cache volume: ${cacheNamespace}_*`);
+      await volumePrune({ renovate: cacheNamespace });
+    } else if (dockerCache === 'mount') {
+      const cacheRoot = join(cacheDir, cacheNamespace);
+      if (await exists(cacheRoot)) {
+        logger.trace(`Deleting Docker cache directory: ${cacheRoot}`);
+        try {
+          await remove(cacheRoot);
+        } catch (err) {
+          await fixPermissionsBeforeDelete(cacheRoot);
+          await remove(cacheRoot);
+        }
+      }
+    }
+  }
+}
+
+export async function ensureDockerTmpCache(): Promise<void> {
```

Comment:
Can these new functions be split into a new file?

Reply:
Done